### PR TITLE
vrt_vcl: Minor polish of director code

### PR DESCRIPTION
**Pull request by dridi**
I was in the neighborhood for a quick loop polish and after a couple rounds of confusion I put my finger on a race and an unsafe sequence of events. I think I fixed one and have a suggestion to fix the other one.

It probably needs more homework.

### CODE REVIEW

**Comment by nigoroll** [comment]
#3949 ?

File: bin/varnishd/cache/cache_vrt_vcl.c [hunk 1]
```diff
@@ -336,6 +336,8 @@ VRT_Assign_Backend(VCL_BACKEND *dst, VCL_BACKEND src)
 	AN(dst);
 	CHECK_OBJ_ORNULL((*dst), DIRECTOR_MAGIC);
 	CHECK_OBJ_ORNULL(src, DIRECTOR_MAGIC);
+	if (*dst == src)
```
Comment (nigoroll) [inline bin/varnishd/cache/cache_vrt_vcl.c:339]: I think you should just push this one to master

Comment (dridi) [reply bin/varnishd/cache/cache_vrt_vcl.c:339]: b825f55f348595cfc24f5eee92b86ecf2e880924

**Comment by dridi** [comment]
> #3949 ?

Apologies, I was not aware of this issue. I'm closing this pull request as my goal was initially to make the loop more straightforward before I noticed the dst==src case. I'll leave it alone since there may be activity in this area related to #3949.

**Pull request closed without merging**